compiler/src/dotty/tools/dotc/typer/EtaExpansion.scala

```diff
@@ -157,6 +157,30 @@ class LiftComplex extends Lifter {
 }
 object LiftComplex extends LiftComplex
 
+/** Lift complex + lift the prefixes */
+object LiftCoverage extends LiftComplex {
+
+  var liftEverything = false
+
+  /** Return true if the apply needs a lift in the coverage phase
+    Return false if the args are empty, if one or more will be lifter by a
+    complex lifter.
+   */
+  def needsLift(tree: tpd.Apply)(using Context): Boolean =
+    !tree.args.isEmpty && !tree.args.forall(super.noLift(_))
```

Comment:
I wonder if this is something that really belongs here as it only depends on `LiftComplex`. It feels like it should be a private method of `CoverageTransormer`.